framework/doc/content/syntax/Functors/index.md

```diff
@@ -0,0 +1,94 @@
+# Functor system
+
+Functors are an abstraction, existing as a base class, that is available to numerous systems in MOOSE:
+
+- [Variables](syntax/Variables/index.md)
+- [Auxiliary variables](syntax/AuxVariables/index.md)
+- [Functor material properties](syntax/FunctorMaterials/index.md)
+- [Functions](syntax/Functions/index.md)
+
+All functors can be called using the same interfaces. This enables considerable code re-use.
+For example, instead of having a kernel for each type of coupled forcing term, like
+[CoupledForce.md], [BodyForce.md], [MatCoupledForce.md], [ADMatCoupledForce.md], we could just have a single object
+`FunctorForce.md` and have the force term be a functor.
+
+`Functions` provide a good analogy to `Functors`. `Functions` are evaluated on-the-fly at a location in space and time.
+`Functors` are evaluated on-the-fly with a space and a time argument. The space arguments can be an element, a point in an element,
+a face of an element, and so on. The time arguments represent the state of the functor : current, previous value (whether in time or iteration),
+or value before that previous one.
+
+## Spatial arguments to functors
+
+In the following subsections, we describe the various spatial arguments that functor can be evaluated at.
+Almost no functor developers should have to concern
+themselves with these details as most functor definitions should just appear as functions of
+space and time, e.g. the same lambda defining the property evaluation should apply across all
+spatial and temporal arguments. However, in the case that a functor developer wishes to
+create specific implementations for specific arguments (as illustrated in `IMakeMyOwnFunctorProps`
+test class) or simply wishes to know more about the system, we give the details below.
+
+Any call to a functor looks like the following
+`_foo(const SpatialArg & r, const TemporalArg & t)`. Below are the possible type overloads of
+`SpatialArg`.
+
+### FaceArg id=spatial-overloads
```

Comment:
Need to update all of these. I can make a pull request against your branch if you want some help sharing the load here

Reply:
can do it